### PR TITLE
Enhancement: Binary Logic: Add uppercase OR, XOR, AND, NOT operators

### LINKED ISSUE #2932: Binary Logic: Does not support capital alphabet
This IA does not give answer for  `3 AND 5` .

I was wondering if we can support capital letters as well.

IA Page:  http://duck.co/ia/view/binary_logic 

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @Sloff

### PR DESCRIPTION
**Pull request by msingle**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [ ] New Instant Answer
    - [ ] Cheat Sheets: **`New {Cheat Sheet Name} Cheat Sheet`**
    - [ ] Other: **`New {IA Name} Instant Answer`**
- [x ] Improvement
    - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
    - [ x] Enhancement: **`{Binary Logic}: {Add uppercase operators}`**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**

###### Description of changes

Adds uppercase AND, OR, XOR, NOT operators

###### Which issues (if any) does this fix?

Fixes #2932 

###### People to notify (@mention interested parties)

---

Instant Answer Page: https://duck.co/ia/view/binary_logic

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @Sloff

### CODE REVIEW

**Comment by daxtheduck** [comment]
@msingle Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/BinaryLogic.pm [hunk 1]
```diff
@@ -10,8 +10,10 @@ use Marpa::R2;
 # Regexp triggers are used to find cases where the logical symbol
 # for 'not' is at the beginning of the query (e.g. the case '¬1')
 triggers query_raw => qr/.*\s+(and|or|xor)\s+.*/;
+triggers query_raw => qr/.*\s+(AND|OR|XOR)\s+.*/;
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/BinaryLogic.pm]: I left some comments on #3114 that will also apply here - you should have a quick look at those and relate them to this work.

**Comment by msingle** [comment]
@GuiltyDolphin, I don't understand what you mean about the 'pass the lowercase string to the parser.' If there is something else I need to do, I'd be happy to do it.

File: lib/DDG/Goodie/BinaryLogic.pm [hunk 2]
```diff
@@ -27,9 +27,13 @@ Term ::=
        ('(') Term (')') assoc => group action => ::first
      | Number
      | 'not' Term action => do_not
+     | 'NOT' Term action => do_not
     || Term 'xor' Term action => do_xor
+    || Term 'XOR' Term action => do_xor
      | Term 'and' Term action => do_and
+     | Term 'AND' Term action => do_and
      | Term 'or' Term action => do_or
+     | Term 'OR' Term action => do_or
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/BinaryLogic.pm]: We can probably get rid of the uppercase variants and just use the `query_lc` handler.

**Comment by msingle** [comment]
@GuiltyDolphin, anything else?

**Comment by GuiltyDolphin** [comment]
@msingle This looks good to me, just waiting for sign-off from @Sloff!

**Comment by mintsoft** [comment]
@GuiltyDolphin Timeout; your call

**Comment by GuiltyDolphin** [comment]
@mintsoft @msingle Thanks! Awesome work! :dolphin:

**Pull request merged**